onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -332,8 +332,9 @@ def add_input(
         return tensor_value  # type: ignore[return-value]
 
     @beartype
-    def add_initializer(self, input_name: str, input_value: torch.Tensor) -> None:
+    def add_initializer(self, input_name: str, input_value: torch.Tensor) -> TorchScriptTensor:
         self._initializers[input_name] = input_value
+        return self.add_input(input_name, input_value)
```

Comment:
Looking at this again. It would make more sense to keep this code and do the "initializer and input" refactoring later when we have a dedicated onnx IR, than doing it on torch script exporter `export_onnx`. I think a comment here for clarification would be nice.

It is also a good opportunity to clean up related code in exporter side. https://github.com/pytorch/pytorch/blob/b97ce3774a403444d21845b0f42e2b98d8faa0f3/torch/onnx/_internal/fx/passes/fx_to_onnxscript.py#L360-L382

At least 3 issues:
1. Initializers are duplicated every time for a `get_attr` on same parameter.
2. `add_input` call separately outside of `add_initializer`.
3. `current_attr` type confusion..

Reply:
On a higher level, this requirement means you are operating on 'TorchScript' IR level. If you are operating on ModelProto you won't need this API. Curious if there is a particular need for the process to happen on 'TorchScript' graph? @wschin